lib/posix/options/Kconfig.xsi_single_process

```diff
@@ -0,0 +1,14 @@
+# Copyright (c) 2024 Tenstorrent AI ULC
+#
+# SPDX-License-Identifier: Apache-2.0
+
+config XSI_SINGLE_PROCESS
+	bool "X/Open single process"
+	depends on POSIX_SINGLE_PROCESS
+	depends on POSIX_TIMERS
```

Comment:
I think that XSI_SINGLE_PROCESS should select these other options, similarly to how it's done for XSI_REALTIME. That's a change for another PR though.